runtime/queries/metricsview.go

```diff
@@ -292,6 +292,327 @@ func buildFilterClauseForCondition(mv *runtimev1.MetricsViewSpec, cond *runtimev
 	return fmt.Sprintf("AND (%s) ", condsClause), args, nil
 }
 
+type identifier struct {
+	mapped string
+	unnest bool
+}
+
+func newIdentifier(name string) identifier {
+	return identifier{safeName(name), false}
+}
+
+func dimensionAliases(mv *runtimev1.MetricsViewSpec) map[string]identifier {
+	aliases := map[string]identifier{}
+	for _, dim := range mv.Dimensions {
+		aliases[dim.Name] = identifier{safeName(metricsViewDimensionColumn(dim)), dim.Unnest}
+	}
+	return aliases
+}
```

Comment:
The previous `buildFilterClauseForMetricsViewFilter` implementation avoided building a lookup map since the overhead of building it (allocating/assigning a ~20 element map, plus a similar number of allocations for `safeName`) would be much higher than just iterating over `mv.Dimensions` a few times (most filters apply to just one or two dimensions).

Always hard to balance optimization vs. convenience, and not sure how much this matters, but would consider staying with the format of `buildFilterClauseForMetricsViewFilter` (i.e. passing `mv` and aliases directly as args; and maybe use a boolean like `isHaving` to indicate which to use for lookups).

Reply:
Ya makes sense. We should really look at having some global cache per metrics view. Things would be very clean with a map. But for now i have updated to use mv and aliases